**Data frame #1 (salary data)**

In [3]:
# Q1: Loading data #1
import pandas as pd
dfi = pd.read_csv (r"https://raw.githubusercontent.com/data602sps/datasetspractice/main/salarydata.csv")

In [4]:
# Q2: Preview the data:
print(dfi)

                   Sate  2018 Mean Annual Wage  Unnamed: 2
0               Alabama                 90,620         NaN
1                 Alaska                99,180         NaN
2                Arizona                88,800         NaN
3               Arkansas                81,430         NaN
4             California                97,110         NaN
5               Colorado                82,560         NaN
6            Connecticut                96,010         NaN
7               Delaware                93,880         NaN
8   District of Columbia                89,880         NaN
9                Florida                87,410         NaN
10               Georgia                86,320         NaN
11                Hawaii                90,540         NaN
12                 Idaho                77,700         NaN
13              IIIinois                90,690         NaN
14               lndiana                83,680         NaN
15                  lowa                82,960         N

There are four columns including the index; state, 2018 mean annual wage and unnamed:2. There are 51 rows plus the header.

In [5]:
# Q3: the shape of the data and the type of the columns:
print(dfi.shape)
print(dfi.dtypes)

(51, 3)
Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object


In [6]:
# Q4: What does .describe() return?
dfi.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

The last column of the data has the same value which is "NaN" but the type is float.
.describe() command returns the statistics measures of the column of the data that is float.

In [7]:
# Q5: What do the .head() and .tail() command do?
dfi.head()

Sate  2018 Mean Annual Wage  Unnamed: 2
0    Alabama                 90,620         NaN
1      Alaska                99,180         NaN
2     Arizona                88,800         NaN
3    Arkansas                81,430         NaN
4  California                97,110         NaN

In [8]:
dfi.tail()

Sate  2018 Mean Annual Wage  Unnamed: 2
46       Virginia                91,700         NaN
47     Washington                85,930         NaN
48  West Virginia                89,420         NaN
49      Wisconsin                85,200         NaN
50        Wyomimg                87,510         NaN

The .head() command returns the first few rows of the data. The .tail() command returns the last few rows of the data.



**Data frame #2 (weather)**

In [9]:
# Q1: Loading data #2
dfii = pd.read_csv (r"https://raw.githubusercontent.com/data602sps/datasetspractice/main/weather.csv")

In [10]:
# Q2: Preview the data:
print(dfii)

      Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
0              1    EWR  2013    1.0   1.0   0.0  37.04  21.92  53.97   
1              2    EWR  2013    1.0   1.0   1.0  37.04  21.92  53.97   
2              3    EWR  2013    1.0   1.0   2.0  37.94  21.92  52.09   
3              4    EWR  2013    1.0   1.0   3.0  37.94  23.00  54.51   
4              5    EWR  2013    1.0   1.0   4.0  37.94  24.08  57.04   
...          ...    ...   ...    ...   ...   ...    ...    ...    ...   
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
0        230.0    10.35702  11.918651     0.0    1013.9   

There are 15 column excluding the index and there are 8719 rows.

In [11]:
# Q3: the shape of the data and the type of the columns:
print(dfii.shape)
print(dfii.dtypes)

(8719, 15)
Unnamed: 0      int64
origin         object
year            int64
month         float64
day           float64
hour          float64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
dtype: object


In [12]:
# Q4: What does .describe() return?
dfii.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

.describe command returns the statstics measures of all columns that are float

In [13]:
# Q5: What do the .head() and .tail() command do?
dfii.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [14]:
dfii.tail()

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0

The .head() command returns the first five rows of the data. The .tail() command returns the last five rows of the data.


**Data frame #3 (2012 SAT results)**

In [15]:
# Q1: Loading data #3
dfiii = pd.read_csv (r"https://raw.githubusercontent.com/SalouaDaouki/Data602/main/2012_SAT_Results.csv")

In [16]:
# Q2: Preview the data:
print(dfiii)

        DBN                                    SCHOOL NAME  \
0    01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1    01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2    01M450                     EAST SIDE COMMUNITY SCHOOL   
3    01M458                      FORSYTH SATELLITE ACADEMY   
4    01M509                        MARTA VALLE HIGH SCHOOL   
..      ...                                            ...   
473  75X012               P.S. X012 LEWIS AND CLARK SCHOOL   
474  75X754        J. M. RAPPORT SCHOOL CAREER DEVELOPMENT   
475  79M645     SCHOOL FOR COOPERATIVE TECHNICAL EDUCATION   
476  79Q950                            GED PLUS s CITYWIDE   
477  79X490                                PHOENIX ACADEMY   

    Num of SAT Test Takers SAT Critical Reading Avg. Score  \
0                       29                             355   
1                       91                             383   
2                       70                             377   
3      

There are 6 columns excluding the index. There are 478 rows.

In [17]:
# Q3: the shape of the data and the type of the columns:
print(dfiii.shape)
print(dfiii.dtypes)

(478, 6)
DBN                                object
SCHOOL NAME                        object
Num of SAT Test Takers             object
SAT Critical Reading Avg. Score    object
SAT Math Avg. Score                object
SAT Writing Avg. Score             object
dtype: object


In [18]:
# Q4: What does .describe() return?
dfiii.describe()

DBN                                    SCHOOL NAME  \
count      478                                            478   
unique     478                                            478   
top     01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
freq         1                                              1   

       Num of SAT Test Takers SAT Critical Reading Avg. Score  \
count                     478                             478   
unique                    175                             164   
top                         s                               s   
freq                       57                              57   

       SAT Math Avg. Score SAT Writing Avg. Score  
count                  478                    478  
unique                 173                    163  
top                      s                      s  
freq                    57                     57

.describe() command returns here different information since all columns are of type object. The output was the count and unique which both have the same value. Then top, which, I think, it is the highest value in the data. And the frequency, which is, maybe, the frequency of the top value.

In [19]:
# Q5: What do the .head() and .tail() command do?
dfiii.head()

DBN                                    SCHOOL NAME  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2  01M450                     EAST SIDE COMMUNITY SCHOOL   
3  01M458                      FORSYTH SATELLITE ACADEMY   
4  01M509                        MARTA VALLE HIGH SCHOOL   

  Num of SAT Test Takers SAT Critical Reading Avg. Score SAT Math Avg. Score  \
0                     29                             355                 404   
1                     91                             383                 423   
2                     70                             377                 402   
3                      7                             414                 401   
4                     44                             390                 433   

  SAT Writing Avg. Score  
0                    363  
1                    366  
2                    370  
3                    359  
4                    384

In [20]:
dfiii.tail()

DBN                                 SCHOOL NAME  \
473  75X012            P.S. X012 LEWIS AND CLARK SCHOOL   
474  75X754     J. M. RAPPORT SCHOOL CAREER DEVELOPMENT   
475  79M645  SCHOOL FOR COOPERATIVE TECHNICAL EDUCATION   
476  79Q950                         GED PLUS s CITYWIDE   
477  79X490                             PHOENIX ACADEMY   

    Num of SAT Test Takers SAT Critical Reading Avg. Score  \
473                      s                               s   
474                      s                               s   
475                      s                               s   
476                      8                             496   
477                      9                             367   

    SAT Math Avg. Score SAT Writing Avg. Score  
473                   s                      s  
474                   s                      s  
475                   s                      s  
476                 400                    426  
477                 370                    360

The head and tail commands still return the first and lastfive rows, respectively,  in the data no matter how big or small the data is.

**Data frame #4 (Leading Causes of death in NYC)**

In [21]:
# Q1: Loading data #4
dfvi = pd.read_csv (r"https://raw.githubusercontent.com/SalouaDaouki/Data602/main/New_York_City_Leading_Causes_of_Death.csv")

In [22]:
# Q2: Preview the data:
print(dfvi)

      Year                                      Leading Cause   Sex  \
0     2011       Chronic Lower Respiratory Diseases (J40-J47)     M   
1     2008              Malignant Neoplasms (Cancer: C00-C97)     M   
2     2010     Chronic Liver Disease and Cirrhosis (K70, K73)     M   
3     2014              Malignant Neoplasms (Cancer: C00-C97)     M   
4     2014  Certain Conditions originating in the Perinata...     M   
...    ...                                                ...   ...   
1798  2017  Chronic Liver Disease and Cirrhosis (K70, K73-...  Male   
1799  2017               Aortic Aneurysm and Dissection (I71)  Male   
1800  2017                          Alzheimer's Disease (G30)  Male   
1801  2017                               Septicemia (A40-A41)  Male   
1802  2017                                   All Other Causes  Male   

             Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
0        White Non-Hispanic    444       33.3                    27.8  
1  

There are 7 columns, and there are 1803 rows.

In [23]:
# Q3: the shape of the data and the type of the columns:
print(dfvi.shape)
print(dfvi.dtypes)

(1803, 7)
Year                        int64
Leading Cause              object
Sex                        object
Race Ethnicity             object
Deaths                     object
Death Rate                 object
Age Adjusted Death Rate    object
dtype: object


In [24]:
# Q4: What does .describe() return?
dfvi.describe()

Year
count  1803.000000
mean   2013.047698
std       3.767515
min    2007.000000
25%    2010.000000
50%    2013.000000
75%    2016.000000
max    2019.000000

In this case, .describe() command returns the statistics measurent for the year only since that the only column that contains integers.

In [25]:
# Q5: What do the .head() and .tail() command do?
dfvi.head()

Year                                      Leading Cause Sex  \
0  2011       Chronic Lower Respiratory Diseases (J40-J47)   M   
1  2008              Malignant Neoplasms (Cancer: C00-C97)   M   
2  2010     Chronic Liver Disease and Cirrhosis (K70, K73)   M   
3  2014              Malignant Neoplasms (Cancer: C00-C97)   M   
4  2014  Certain Conditions originating in the Perinata...   M   

       Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
0  White Non-Hispanic    444       33.3                    27.8  
1  White Non-Hispanic   3361      250.4                   214.3  
2            Hispanic    150       13.1                    17.6  
3            Hispanic   1146         96                   143.5  
4  Not Stated/Unknown     25          .                       .

In [26]:
dfvi.tail()

Year                                      Leading Cause   Sex  \
1798  2017  Chronic Liver Disease and Cirrhosis (K70, K73-...  Male   
1799  2017               Aortic Aneurysm and Dissection (I71)  Male   
1800  2017                          Alzheimer's Disease (G30)  Male   
1801  2017                               Septicemia (A40-A41)  Male   
1802  2017                                   All Other Causes  Male   

             Race Ethnicity Deaths Death Rate Age Adjusted Death Rate  
1798  Other Race/ Ethnicity      2        NaN                     NaN  
1799  Other Race/ Ethnicity      2        NaN                     NaN  
1800  Other Race/ Ethnicity      2        NaN                     NaN  
1801  Other Race/ Ethnicity      2        NaN                     NaN  
1802  Other Race/ Ethnicity     16        NaN                     NaN

**Extra Credits**:
1. Choose one of your datasets and remove the header information. (Can delete the row in excel, etc..)
2. Import the data into your environment using pandas. Display the .head() of your data showing no header information.
3. Using pandas, update the dataset to include the header information. Display the updated data using .head().

In [27]:
dfv = pd.read_csv(r"https://raw.githubusercontent.com/SalouaDaouki/Data602/main/2012_SAT_Results_noHeader.csv", header=None)
dfv.head()

0                                              1   2    3    4    5
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES  29  355  404  363
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL  91  383  423  366
2  01M450                     EAST SIDE COMMUNITY SCHOOL  70  377  402  370
3  01M458                      FORSYTH SATELLITE ACADEMY   7  414  401  359
4  01M509                        MARTA VALLE HIGH SCHOOL  44  390  433  384

In [28]:
headers = ["DBN", "School_Name", "Num_of_test_Takers", "SAT_critical_Reading_Avg._score", "SAT_Math_Avg._score", "Sat_Writing_Avg._score"]
dfv.columns = headers
dfv.head()

DBN                                    School_Name Num_of_test_Takers  \
0  01M292  HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES                 29   
1  01M448            UNIVERSITY NEIGHBORHOOD HIGH SCHOOL                 91   
2  01M450                     EAST SIDE COMMUNITY SCHOOL                 70   
3  01M458                      FORSYTH SATELLITE ACADEMY                  7   
4  01M509                        MARTA VALLE HIGH SCHOOL                 44   

  SAT_critical_Reading_Avg._score SAT_Math_Avg._score Sat_Writing_Avg._score  
0                             355                 404                    363  
1                             383                 423                    366  
2                             377                 402                    370  
3                             414                 401                    359  
4                             390                 433                    384

**Extra Credits**
1. Import a “dirty” dataset from the internet into your environment. (Missing values, improper coding of columns, etc.)
2. Clean or “tidy” the data when loading into python using pandas.

In [29]:
dfiv = pd.read_csv(r"https://raw.githubusercontent.com/SalouaDaouki/Data602/main/Clean_Air_Tracking_System__CATS__Permits.csv")
dfiv.head()

<ipython-input-29-7565f092157c>:1: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  dfiv = pd.read_csv(r"https://raw.githubusercontent.com/SalouaDaouki/Data602/main/Clean_Air_Tracking_System__CATS__Permits.csv")


RequestID ApplicationID             RequestType House             Street  \
0     113975      CA000066             WORK PERMIT   NaN                NaN   
1     133681      CA000163  CERTIFICATE TO OPERATE    34     WEST 17 STREET   
2      51758      CA000182            REGISTRATION   102           AVENUE S   
3     120600      CA000183  CERTIFICATE TO OPERATE  1295      MORRIS AVENUE   
4      53528      CA000185            REGISTRATION   291  CENTRAL PARK WEST   

     Borough        Bin   Block     Lot                      OwnerName  ...  \
0      BRONX  3000000.0    36.0  3118.0                  LYDIA SPECTOR  ...   
1  MANHATTAN  1015401.0   818.0  7502.0            F M RING ASSOCIATES  ...   
2   BROOKLYN  3189913.0  7079.0     3.0           DONAHUE, PATRICIA A.  ...   
3      BRONX  2007805.0  2816.0    70.0           MORRISANIA IV ASSOC.  ...   
4  MANHATTAN  1031567.0  1203.0  7502.0  DOUGLAS,ELLIMAN,GIBBONS&AMP;I  ...   

                               Make                             Model  \
0                 ROCKMILLS  MP-090                 ROCKMILLS  MP-090   
1                 FEDERAL 003-562LB                 FEDERAL 003-562LB   
2  WEIL MCLAIN              EGH 125  WEIL MCLAIN              EGH 125   
3                  RAYPACK EA 2001T                  RAYPACK EA 2001T   
4                       WEIL MCLAIN                             768-S   

                          BurnerMake                        BurnerModel  \
0                     HEVEOIL  AM 6E                     HEVEOIL  AM 6E   
1                      ICI # AM412DH                      ICI # AM412DH   
2  WEIL MCLAIN              INTEGRAL  WEIL MCLAIN              INTEGRAL   
3                   RAYPACK EA 2001T                   RAYPACK EA 2001T   
4                       GORDON PIATT                           WR8-1-07   

  PrimaryFuel SecondaryFuel Quantity               IssueDate           status  \
0     NO6FUEL          NONE      1.0  08/14/1989 12:00:00 AM  CANCELLED         
1     NO4FUEL          NONE      1.0  01/01/1900 12:00:00 AM  CANCELLED         
2  NATURALGAS          NONE      1.0  01/04/1982 12:00:00 AM  EXPIRED           
3  NATURALGAS          NONE      2.0  10/24/1986 12:00:00 AM  CANCELLED         
4     NO2FUEL          NONE      1.0  01/25/1988 12:00:00 AM  EXPIRED           

                PremiseName  
0                       NaN  
1                       NaN  
2                       NaN  
3                       NaN  
4  SUPERSEDED BY CA0276/94K  

[5 rows x 21 columns]

In [38]:
# grouping the data by borough
dfiv1= dfiv.groupby(['Borough', 'status'])
dfiv1.first()

RequestID ApplicationID             RequestType  \
Borough      status                                                             
BRONX        CANCELLED           113975      CA000066             WORK PERMIT   
             CURRENT             326439      CA000188  CERTIFICATE TO OPERATE   
             EXPIRED             353235      CA000591            REGISTRATION   
BROOKLYN     CANCELLED           126941      CA000186  CERTIFICATE TO OPERATE   
             CURRENT             312993      CA000282            REGISTRATION   
             EXPIRED              51758      CA000182            REGISTRATION   
MANHATTAN    CANCELLED           133681      CA000163  CERTIFICATE TO OPERATE   
             CURRENT             376262      CA000296            REGISTRATION   
             EXPIRED              53528      CA000185            REGISTRATION   
NONE         CURRENT             380558      PB014614            REGISTRATION   
             EXPIRED             156581      PB038414            REGISTRATION   
QUEENS       CANCELLED           121120      CA000273  CERTIFICATE TO OPERATE   
             CURRENT             341363      CA000589            REGISTRATION   
             EXPIRED             293691      CA000390  CERTIFICATE TO OPERATE   
STATENISLAND CANCELLED           119614      CA000878             WORK PERMIT   
             CURRENT             325146      CA002077            REGISTRATION   
             EXPIRED              59837      CA000896            REGISTRATION   

                                House               Street        Bin   Block  \
Borough      status                                                             
BRONX        CANCELLED           1295        MORRIS AVENUE  3000000.0    36.0   
             CURRENT             2455        CRUGER AVENUE  2051308.0  4430.0   
             EXPIRED             1087        SUMMIT AVENUE  2003491.0  2526.0   
BROOKLYN     CANCELLED            380     EAST   18 STREET  3118770.0  5160.0   
             CURRENT              701         RALPH AVENUE  3080793.0  3509.0   
             EXPIRED              102             AVENUE S  3189913.0  7079.0   
MANHATTAN    CANCELLED             34       WEST 17 STREET  1015401.0   818.0   
             CURRENT              616      WEST 113 STREET  1057341.0  1895.0   
             EXPIRED              291    CENTRAL PARK WEST  1031567.0  1203.0   
NONE         CURRENT          VARIOUS            LOCATIONS  3335138.0   14260   
             EXPIRED              400  ROOSEVELT AVE (VAR)  3000000.0     0.0   
QUEENS       CANCELLED          94-40           118 STREET  3000000.0  9435.0   
             CURRENT            25-63            33 STREET  4009279.0   631.0   
             EXPIRED           210-50            41 AVENUE  4137452.0  6227.0   
STATENISLAND CANCELLED           3000     ARTHUR KILL ROAD  3000000.0  7247.0   
             CURRENT               62       CLOISTER PLACE  5052430.0  3630.0   
             EXPIRED              460       BRIELLE AVENUE  5141697.0   955.0   

                                 Lot                           OwnerName  \
Borough      status                                                        
BRONX        CANCELLED        3118.0                       LYDIA SPECTOR   
             CURRENT            28.0             2455 CURGBER AVE. CORP.   
             EXPIRED            70.0                        PROMESA HDFC   
BROOKLYN     CANCELLED           1.0                         IVAN I FIST   
             CURRENT             1.0                       SUTTER HOUSES   
             EXPIRED             3.0                DONAHUE, PATRICIA A.   
MANHATTAN    CANCELLED        7502.0                 F M RING ASSOCIATES   
             CURRENT            30.0                        TOM ETTINGER   
             EXPIRED          7502.0       DOUGLAS,ELLIMAN,GIBBONS&AMP;I   
NONE         CURRENT             1.0                   VARIOUS LOCATIONS   
             EXPIRED             0.0  